pkg/build/api/types.go

```diff
@@ -396,11 +408,9 @@ type GitBuildSource struct {
 	// Ref is the branch/tag/ref to build.
 	Ref string
 
-	// HTTPProxy is a proxy used to reach the git repository over http
-	HTTPProxy *string
-
-	// HTTPSProxy is a proxy used to reach the git repository over https
-	HTTPSProxy *string
+	// ProxyConfig defines the proxies to use for the git clone operation
+	// +k8s:conversion-gen=false
+	ProxyConfig ProxyConfig
```

Comment:
Sorry, I should have been clearer.  I expected this to look like `CommonSpec` and have the same config internal and external with the correct inlining statements.  Is the only reason you have conversion-gen=false different is because of that?  If so my intent was just to have:

```
type GitBuildSource struct {
  ...
  GitProxyConfig `json:",inline"`
}

type GitProxyConfig struct {
  HTTPProxy *string
  ...
}
```